File: backend/src/main/java/com/biotrace/service/HospitalService.java
package com.biotrace.service;

import com.biotrace.exception.ResourceNotFoundException;
import com.biotrace.model.Hospital;
import com.biotrace.repository.HospitalRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class HospitalService {

    @Autowired
    private HospitalRepository hospitalRepository;

    public List<Hospital> getAllHospitals() {
        return hospitalRepository.findAll();
    }

    public Hospital getHospitalById(Long id) {
        return hospitalRepository.findById(id)
            .orElseThrow(() -> new ResourceNotFoundException("Hospital not found"));
    }

    public Hospital createHospital(Hospital hospital) {
        return hospitalRepository.save(hospital);
    }

    public Hospital updateHospital(Long id, Hospital hospitalDetails) {
        Hospital hospital = getHospitalById(id);
        hospital.setName(hospitalDetails.getName());
        hospital.setAddress(hospitalDetails.getAddress());
        hospital.setContactPerson(hospitalDetails.getContactPerson());
        hospital.setPhone(hospitalDetails.getPhone());
        hospital.setEmail(hospitalDetails.getEmail());
        hospital.setRegistrationNumber(hospitalDetails.getRegistrationNumber());
        return hospitalRepository.save(hospital);
    }

    public void deleteHospital(Long id) {
        Hospital hospital = getHospitalById(id);
        hospitalRepository.delete(hospital);
    }
}

File: frontend/src/services/hospitalService.js
import api from './api';

export const hospitalService = {
  getAllHospitals: async () => {
    const response = await api.get('/hospitals');
    return response.data;
  },

  getHospitalById: async (id) => {
    const response = await api.get(`/hospitals/${id}`);
    return response.data;
  },

  createHospital: async (hospitalData) => {
    const response = await api.post('/hospitals', hospitalData);
    return response.data;
  },

  updateHospital: async (id, hospitalData) => {
    const response = await api.put(`/hospitals/${id}`, hospitalData);
    return response.data;
  },

  deleteHospital: async (id) => {
    const response = await api.delete(`/hospitals/${id}`);
    return response.data;
  },

  getHospitalStats: async () => {
    const response = await api.get('/hospitals/stats');
    return response.data;
  }
};

export default hospitalService;
